### PR TITLE
[WFLY-11340] Fix mp ot tests security manager

### PR DESCRIPTION
**Pull request by jstourac**
https://issues.jboss.org/browse/WFLY-11340

As commented in linked issue, I am not sure about proper permission for the last test - `SimpleRestClientTestCase`. Any suggestion is welcomed.

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:
- [x] Pull Request title is properly formatted: `[WFLY-XYZ] Subject` or `WFLY-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue(s)
- [ ] Pull Request contains description of the issue(s)
  - too complicated description, can be seen in jira anyway
- [ ] Pull Request does not include fixes for issues other than the main ticket
  - yes and no - there is an extra commit that adds license headers into the modified source files... I don't think it is worth to create separate issue for it though...
- [x] Attached commits represent units of work and are properly formatted

### CODE REVIEW

**Comment by jstourac** [comment]
Retest this, please.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/microprofile/opentracing/ResourceNotTracedTestCase.java [hunk 1]
```diff
@@ -30,6 +53,22 @@
     @ArquillianResource
     private URL url;
 
+    /**
+     * Permissions required in case security manager is enabled.
+     */
+    private static final String PERMISSIONS = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
```
Comment (jamezp) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/microprofile/opentracing/ResourceNotTracedTestCase.java]: There is a [utility](https://github.com/wildfly/wildfly/blob/bf07fdae275a8fd208d483d4670324fd28c2d88f/testsuite/shared/src/main/java/org/jboss/as/test/shared/integration/ejb/security/PermissionUtils.java#L42) for this. Have a look at this [example](https://github.com/wildfly/wildfly/blob/4eaf7e53290b5e6a0c84b78efbafaff7b014b87c/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/batch/chunk/ChunkPartitionTestCase.java#L74-L79).

**Review by jamezp — COMMENTED** [review]
No need to make the immediate changes yet. I want to have a better look at this as ideally we don't ever need to add permissions like this to a test case. Obviously in some control cases that is a requirement, but I don't think in this case we want that. Again though I need to take a better look :)

**Comment by jstourac** [comment]
@jamezp, Thank you for your review. I'll wait for your further investigation then. Let me know when you think I should amend this anyhow or want some information from me.

**Comment by bstansberry** [comment]
The java.security.AllPermission in the one test is the more troublesome one.  

The others it seems the app is trying to open an HTTP client, so it's normal that permissions would be needed for that and the need is purely due to the behavior of the app; i.e. the app author's own decision.  Setting that permission up in the optimal way is good, but the need for the permission isn't worrisome.

The AllPermission one though, it seems odd that a call to javax.ws.rs.client.ClientBuilder.newBuilder would require complex permissions, and granting AllPermission seems like it might be covering over a container problem.

**Comment by jamezp** [comment]
I'm closing this as I created #11890. It's essentially the same only using the API and restricting the access in the `SimpleRestClientTestCase` to a smaller set.

**Pull request closed without merging**